tests/scripts/local_monitoring_tests/Local_monitoring_tests.md

```diff
@@ -0,0 +1,96 @@
+# **Kruize Local monitoring tests**
+
+Kruize Local monitoring tests validates the behaviour of [Kruize local monitoring APIs](/design/KruizeLocalAPI.md)
+using various positive and negative scenarios. These tests are developed using pytest framework.
+
+## Tests description
+### **List Datasources API tests**
+
+Here are the test scenarios:
+- List all datasources
+- List datasources with name query parameter:
+    - /datasources?name=<datasource_name>
+- List datasources with invalid parameter value for datasource name tested with empty, NULL or invalid.
+
+### **Import Metadata API tests**
+
+Here are the test scenarios:
+
+- Importing metadata for a valid datasource to the API.
+- Post an invalid header content type
+- Post the same datasource again
+- Test with invalid values such as blank, null or an invalid value for various keys in the dsmetadata input request json
+- Validate error messages when the mandatory fields are missing
+
+The above tests are developed using pytest framework and the tests are run using shell script wrapper that does the following:
+- Deploys kruize in non-CRD mode using the [deploy script](https://github.com/kruize/autotune/blob/master/deploy.sh) from the autotune repo
+- Creates a resource optimization performance profile using the [createPerformanceProfile API](/design/PerformanceProfileAPI.md)
+- Runs the above tests using pytest
+
+## Prerequisites for running the tests:
+- Minikube setup or access to Openshift cluster
+- Tools like kubectl, oc, curl, jq, python
+- Various python modules pytest, json, pytest-html, requests, jinja2
+  (these modules will be automatically installed while the test is run)
+
+## How to run the test?
+
+Use the below command to test :
+
+```
+<KRUIZE_REPO>/tests/test_autotune.sh -c minikube -r [location of benchmarks]  [-i kruize image] [--tctype=functional] [--testmodule=Autotune module to be tested] [--testsuite=Group of tests that you want to perform] [--testcase=Particular test case that you want to test] [-n namespace] [--resultsdir=results directory] [--skipsetup]
+```
+
+Where values for test_autotune.sh are:
+
+```
+usage: test_autotune.sh [ -c ] : cluster type. Supported type - minikube
```

Comment:
Include openshift in supported types. Specify the default option. Please update remote monitoring doc as well.